tests/vision_tests/runVisionTests.py

```diff
@@ -181,35 +179,56 @@ def write_formatted(output, f):
 # Import arguments
 parser = argparse.ArgumentParser()
 parser.add_argument('--runvx_directory',    type=str, default='',
-                    help='RunVX Executable directory - required')
+                    help='RunVX Executable Directory - required')
 parser.add_argument('--hardware_mode',      type=str, default='CPU',
-                    help='OpenVX Vision Function target - optional (default:CPU [options:CPU/GPU])')
+                    help='OpenVX Vision Function Target - optional (default:CPU [options:CPU/GPU])')
+parser.add_argument('--list_tests',         type=str, default='no',
+                    help='List Vision Functionality Tests - optional (default:no [options:no/yes])')
+parser.add_argument('--test_filter',        type=int, default=0,
+                    help='Vision Functionality Test Filter - optional (default:0 [range:1 - N])')
 args = parser.parse_args()
 
 runvxDir = args.runvx_directory
 hardwareMode = args.hardware_mode
+listTest = args.list_tests
+testFilter = args.test_filter
 
 # check arguments
-if runvxDir == '':
-    print(
-        "ERROR: RunVX Executable Directory Required")
-    exit()
 if hardwareMode not in ('CPU', 'GPU'):
     print("ERROR: OpenVX Hardware supported - CPU or GPU]")
     exit()
+if listTest not in ('no', 'yes'):
+    print("ERROR: List Vision Functionality Tests options supported - no or yes]")
+    exit()
+if not 0 <= testFilter <= len(openvxNodes):
+    print(
+        "\nERROR: Vision Functionality Filter not in range - [1 - %d]\n" % (len(openvxNodes)))
+    exit()
+# List Vision Functionality tests
+if listTest == 'yes':
+    print(" %-5s - %-30s\n" % ('Test ID', 'Test Name'))
+    for i in range(len(openvxNodes)):
+        nodeName, nodeFormat = openvxNodes[i]
+        print("   %-5d - %-30s\n" % ((i+1), nodeName))
+    exit()
+if runvxDir == '':
+    print("ERROR: RunVX Executable Directory Required")
+    exit()
 
 # RunVX Application
 runVX_exe = runvxDir+'/runvx'
 runvx_exe_dir = os.path.expanduser(runVX_exe)
 
-print("\nrunVisionTests - OpenVX Vision Tests V-"+__version__+"\n")
 os.system('(cd gdfs; mkdir openvx_test_results)')
-for i in range(len(visionTestConfig)):
-    testFileName = visionTestConfig[i]
-    print("Running Test Script: "+testFileName)
-    os.system('(cd gdfs; ./../'+runvx_exe_dir+' -frames:100 -affinity:' +
-              hardwareMode+' -dump-profile file '+testFileName+' | tee -a openvx_test_results/VisionOutput.log)')
-    print("\n")
+
+if testFilter == 0:
+    print("\nrunVisionTests - OpenVX Vision Tests V-"+__version__+"\n")
+    for i in range(len(visionTestConfig)):
+        testFileName = visionTestConfig[i]
+        print("Running Test Script: "+testFileName)
+        os.system('(cd gdfs; ./../'+runvx_exe_dir+' -frames:100 -affinity:' +
```

Comment:
adding **./../** before runvx_exe_dir is causing runvx not found error. By removing it, I was able to run the script on UB18.04 fine by passing /opt/rocm/mivisionx/bin for the --runvx_directory argument.

Reply:
@asalmanp can you verify if the path to runvx is not /opt/rocm.. but a local dir path, will it raise an issue?

Reply:
@kiritigowda I just verified that if the path to runvx is a local dir path, the original runVisionTests script will cause the same (runvx not found error), and can just be run by removing the ./../ before the runvx_exe_dir..